field/src/field.rs

```diff
@@ -137,6 +137,19 @@ pub trait PrimeCharacteristicRing:
         self.clone() + self.clone()
     }
 
+    /// The elementary function `halve(a) = a/2`.
+    ///
+    /// # Panics
+    /// The function will panic if the field has characteristic 2.
```

Comment:
Nit: Hmm, I'm not sure this code "panics" if the field has characteristic 2, rather this method prevents a field of characteristic 2 implementing the `PrimeCharacteristicRing` trait in the first place. This latter fact should probably appear in the documentation to the `PrimeCharacteristicRing` trait above (which is now arguably an `OddPrimeCharacteristicRing` :slightly_smiling_face:).

Reply:
Well you _could_, though I would not approve that PR. :)